ps_featuredproducts.php

```diff
@@ -263,37 +314,94 @@ public function getWidgetVariables($hookName = null, array $configuration = [])
 
     protected function getProducts()
     {
-        $category = new Category((int) Configuration::get('HOME_FEATURED_CAT'));
-
-        $searchProvider = new CategoryProductSearchProvider(
-            $this->context->getTranslator(),
-            $category
-        );
-
-        $context = new ProductSearchContext($this->context);
-
-        $query = new ProductSearchQuery();
+        $idCategory = (int) Configuration::get('HOME_FEATURED_CAT');
 
         $nProducts = Configuration::get('HOME_FEATURED_NBR');
         if ($nProducts < 0) {
             $nProducts = 12;
         }
 
-        $query
-            ->setResultsPerPage($nProducts)
-            ->setPage(1)
-        ;
+        $nbDaysNewProduct = Configuration::get('PS_NB_DAYS_NEW_PRODUCT');
+        if (!Validate::isUnsignedInt($nbDaysNewProduct)) {
+            $nbDaysNewProduct = 20;
+        }
 
-        if (Configuration::get('HOME_FEATURED_RANDOMIZE')) {
-            $query->setSortOrder(SortOrder::random());
+        $random = !!Configuration::get('HOME_FEATURED_RANDOMIZE');
+        $popularFirst = !!Configuration::get('HOME_FEATURED_POPULAR_FIRST');
+        $onlyAvailable = !!Configuration::get('HOME_FEATURED_ONLY_AVAILABLE');
+
+        $outOfStockGlobal = (int) Configuration::get("PS_ORDER_OUT_OF_STOCK");
+
+        $sql = 'SELECT p.*, o.orders_placed, product_shop.*, stock.out_of_stock, IFNULL(stock.quantity, 0) AS quantity' . (Combination::isFeatureActive() ? ', IFNULL(product_attribute_shop.id_product_attribute, 0) AS id_product_attribute,
```

Comment:
@kelostrada, there's really no way to query everything like this, Presenters handle getting images, etc.

Reply:
Yeah I get it, the query was basically copy pasted from `Category` that is used here by `CategoryProductSearchProvider`. I can clean it up if you want to use this PR. As I said I just wanted to make it work and felt I can share if anyone needs similar functionality

Reply:
thanks @kelostrada 

if you are motivated to finish your PR I'll be happy to accept your contribution, but it has to be reworked to keep the codebase clean 👍🏻